packages/neo-one-client/src/user/keystore/LedgerKeyStore.ts

```diff
@@ -0,0 +1,232 @@
+        );
+
+        return {
+          verification: witness.verification.toString('hex'),
+          invocation: witness.invocation.toString('hex'),
+        };
+      },
+      'neo_ledger_sign',
+      monitor,
+    );
+  }
+
+  public async selectAccount(id?: UserAccountID, _monitor?: Monitor): Promise<void> {
+    if (id === undefined) {
+      this.currentAccountInternal$.next(undefined);
+    } else {
+      const { account } = this.getLedger(id);
+      this.currentAccountInternal$.next(account);
+    }
+  }
+
+  public getLedger({ address, network }: UserAccountID): Ledger {
+    const ledgers = this.ledgers[network];
+    if (ledgers === undefined) {
+      throw new UnknownAccountError(address);
+    }
+
+    const ledger = ledgers[address];
+    if (ledger === undefined) {
+      throw new UnknownAccountError(address);
+    }
+
+    return ledger;
+  }
+
+  public getLedgers$({ address, network }: UserAccountID): Observable<Ledger | undefined> {
+    return this.ledgersInternal$.pipe(
+      map((ledgers) => {
+        const networkLedgers = ledgers[network];
+        if (networkLedgers === undefined) {
+          return undefined;
+        }
+
+        return networkLedgers[address];
+      }),
+    );
+  }
+  // we just init the first account on the ledger for now, ideally we have some kind of scanning method to extract actual accounts.
+  public async init(): Promise<LedgerHandler> {
+    const handler = await LedgerHandler.init();
+    const accounts = await this.scanLedgerAccounts(handler);
+
+    this.initWithLedgers(accounts);
+
+    return handler;
+  }
+
+  public async close(): Promise<void> {
+    const handler = await this.initPromise;
+    await handler.close();
+  }
+
+  private initWithLedgers(accountsToLedgers: ReadonlyArray<Ledger>): void {
+    const ledgers = accountsToLedgers.reduce<Ledgers>(
+      (acc, ledger) => ({
+        ...acc,
+        [ledger.account.id.network]: {
+          ...(acc[ledger.account.id.network] === undefined ? {} : acc[ledger.account.id.network]),
+          [ledger.account.id.address]: ledger,
+        },
+      }),
+      {},
+    );
+
+    this.ledgersInternal$.next(ledgers);
+    this.newCurrentAccount();
+  }
+
+  private newCurrentAccount(): void {
+    const allAccounts = flattenLedgers(this.ledgers).map(({ account: value }) => value);
+
+    const account = allAccounts[allAccounts.length - 2];
+    this.currentAccountInternal$.next(account);
+  }
+
+  private async scanLedgerAccounts(handler: LedgerHandler, maxOffset = 20): Promise<ReadonlyArray<Ledger>> {
+    const scanAccountsInternal = async (start: number, currentOffset = 0): Promise<ReadonlyArray<Ledger>> => {
+      if (currentOffset <= maxOffset) {
+        const publicKey = await handler.getPublicKey(start + currentOffset);
+        const ledgerAccount = this.publicKeyToLedgerAccount(start + currentOffset, publicKey);
+        const account = await this.provider.getAccount(ledgerAccount.account.id.address);
```

Comment:
This is actually a bit tricky - once you change `getAccount` to require a network, you'll need to know which networks to scan here. What you actually want is to take in a provider that looks like
```
interface Provider {
  readonly getNetworks(): ReadonlyArray<NetworkType>;
  readonly networks$: Observable<ReadonlyArray<NetworkType>>;
  readonly getAccount: (network: NetworkType, address: AddressString, monitor?: Monitor) => Promise<Account>;
}
```

(Which happens to match a subset of the `NEOONEProvider` interface)

And then change the `ledgersInternal$` property to look something like 
```
this.ledgers$ = new BehaviorSubject<Ledgers>({});
const ledgersInternal$ = this.provider.networks$.pipe(
  mergeScanLatest((acc, networks) => {
    <filter already scanned "networks" from what we have in "acc.scannedNetworks">
    const next = await Promise.all(networks.map(networksFiltered.map(async (network) =>
      this.scanLedgerAccounts(network, handler);
    )); 
    const nextLedgers = <merge "acc.ledgers" and "next">;
    return { ledgers: nextLedgers, networks: acc.networks.concat(networksFiltered) };
  }),
  map(({ ledgers }) => ledgers),
  publishReplay(1),
);
ledgersInternal$.subscribe(this.ledgers$);
```
and then build the other observable properties off of `this.ledgersInternal$` (i.e. `this.accounts$`)